src/chcanv.cpp

```diff
@@ -6210,16 +6212,20 @@ void ChartCanvas::AlertDraw( ocpnDC& dc )
         }
     } else
         AnchorAlertOn2 = false;
-
+#ifdef HAVE_PORTAUDIO
+    if (++playtimes < g_SoundPlayTime)
+        return;
+    playtimes = 0;
+#endif
```

Comment:
This could be a dynamic_cast test instead, removing evil #ifdefs

Reply:
Sorry No, g_SoudPlaytime has only a value with Portaudio.

Reply:
But it's defined, right? In which case a dynamic_cast check disables it when not using portaudio, but without #ifdefs and corresponding loss of compiler checks.

Reply:
g_SoundPlayTime is a global Variable, Set to 5 somewere as default. But you can only change this value with the Slider in Optiondialog, when Portaudio in enabled. The other Sounds could not.
Set g_SoundPlayTime = 0; .... then it has no effect and the #ifdef's could be deleted. ... But is this better ? 
... as you like !

Reply:
SImple deleting the #ifdef is certainly worse. What I'm talking about is replacing it with

    if (dynamic_cast<PortAudioSound*>(SoundFactory()) {
    ...
    }

which is true if and only if SoundFactory() returns a PortAudioSound.

Reply:
... PortAudioSound* is not known there

Reply:
error C2680: "PortAudioSound *": Ungültiger Zieltyp für dynamic_cast

Reply:
Ah... Deutsch! :)

It's defined in PortAudioSound.h. You need to include it.